pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -1614,6 +1614,35 @@ private void internalUpdateSubscriptionPropertiesForNonPartitionedTopic(AsyncRes
                 });
     }
 
+    private void internalGetSubscriptionPropertiesForNonPartitionedTopic(AsyncResponse asyncResponse,
+                                                                            String subName,
+                                                                            boolean authoritative) {
+        validateTopicOwnershipAsync(topicName, authoritative)
+                .thenRun(() -> validateTopicOperation(topicName, TopicOperation.CONSUME))
+                .thenApply(__ -> {
+                    Topic topic = getTopicReference(topicName);
```

Comment:
better to use `getTopicReferenceAsync()`?

Reply:
good catch !